### ISSUE TITLE
bad argument #1 to 'set' (expecting number or Tensor or Storage)

### ISSUE THREAD

**Issue by ShowLang** [szagoruyko/loadcaffe#101]
I want load the pretrained model to finetune with my own dataset,so I run those scrips 

`local model = loadcaffe.load('models/vgg16/VGG_ILSVRC_16_layers_deploy.prototxt','logs/vgg16/VGG_ILSVRC_16_layers.caffemodel','cudnn')
print(model)
for i=1,2 do
    model:remove(#model.modules)
end
model:add(nn.Linear(4096,100))
model:add(cudnn.SoftMax())
`
the model loaded successfuly but when I use the function getParameters()whith 

`parameters,gradParameters = model:getParameters()`I got problem that

'/home/ubuntu/git/torch/install/bin/luajit: /home/ubuntu/git/torch/install/share/lua/5.1/nn/Module.lua:286: bad argument #1 to 'set' (expecting number or Tensor or Storage)
stack traceback:
	[C]: in function 'set'
	/home/ubuntu/git/torch/install/share/lua/5.1/nn/Module.lua:286: in function 'flatten'
	/home/ubuntu/git/torch/install/share/lua/5.1/nn/Module.lua:326: in function 'getParameters'
	train.lua:108: in main chunk
	[C]: in function 'dofile'
	.../git/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00405d50
'

**Comment by ShowLang** [comment]
The problem was that the parameters's type  of the last two layers don't match with layers before,so the problem was solved by adding 
`model:add(nn.Linear(4096,100):cuda())
model:add(cudnn.SoftMax():cuda())`
